### PR TITLE
Fix RMT examples

### PR DESCRIPTION
**Pull request by bjoernQ**
- RX examples are now not interactive anymore (need to short two pins) to make them work on ESP32 / ESP32-S2
- use a different frequency for H2 since it cannot reach 80MHz
- use GPIO 4 (and GPIO 5) to avoid conflicting with GPIO 1 and 3 on ESP32
- use GPIOs consistently (i.e. same for async and non-async examples)

### CODE REVIEW

**Review by JurajSadel — APPROVED** [review]
LGTM

File: examples/src/bin/embassy_rmt_rx.rs [hunk 1]
```diff
@@ -12,34 +9,59 @@
...
+        }
+        Timer::after(Duration::from_millis(1000)).await;
+    }
+}
+
 #[main]
-async fn main(_spawner: Spawner) {
+async fn main(spawner: Spawner) {
+    #[cfg(feature = "log")]
+    esp_println::logger::init_logger_from_env();
```
Comment (jessebraham) [inline examples/src/bin/embassy_rmt_rx.rs]: What's the purpose of this line if none of the `log::*` macros are ever called?

Comment (bjoernQ) [reply examples/src/bin/embassy_rmt_rx.rs]: Probably copy-paste from the current original example here: https://github.com/esp-rs/esp-hal/blob/9a95c0aa880af7271f059797339bf890d2d59c64/esp32-hal/examples/embassy_rmt_rx.rs#L41-L42

It makes sense even w/o logging here to e.g. see the PSRAM logs but should get removed

Comment (bjoernQ) [reply examples/src/bin/embassy_rmt_rx.rs]: BTW I think it's also in the I2S examples in your branch .... should I remove them in my other PR then, too?

Comment (jessebraham) [reply examples/src/bin/embassy_rmt_rx.rs]: I had removed them from the examples as they didn't seem to do anything. If you wouldn't mind that'd be great, I guess it's not a huge deal I was just curious why it was added in.

If we _do_ want to include any of these, I think they should at least have a comment stating _why_ they're there.

**Review by jessebraham — APPROVED** [review]
LGTM, thanks!

**Pull request merged**